Adafruit CircuitPython Code of Conduct

kattni opened the pull request:
This is to be the new Adafruit CircuitPython Code of Conduct. It will be displayed on Discord and apply to all CircuitPython spaces. 

Please read through it and provide any suggestions you may have in the form of a review. We want this to be a great fit for our community and the best way to do that is to have community input. Everyone is welcome to submit a review for consideration.

kattni committed:
Initial commit

kattni committed:
First pass at updating for Adafruit Discord

kattni committed:
Updated formatting, added standards content

kattni committed:
Further updates to moderation and reorg of sections

kattni committed:
removed extra space

kattni committed:
Update to scope, wording, and title. Changed license

License changed to Creative Commons Attributions in line with the
Contributor Covenant. Removed standard GitHub license file, added
CC license file.

kattni committed:
Updated formatting, wording in Scope

kattni committed:
Updated README to match master

kattni committed:
Added community values to Moderation

kattni committed:
Added community values to Moderation

tannewt commented on code-of-conduct.md:
```diff
@@ -0,0 +1,165 @@
+asking, instead of providing an answer that covers the question *you think*
+they're trying to ask. If you feel the need to "take the keyboard away" from
+the person requesting assistance, reconsider how you're trying to answer. If
+you're doing all the talking, you may be excessively helping.
```
I don't think this should be an entire paragraph. Its equally important as the "unacceptable behavior" listed below.

tannewt commented on code-of-conduct.md:
```diff
@@ -0,0 +1,165 @@
+On the Adafruit Discord server, please avoid using nicknames or avatar images
+that might detract from a friendly, safe and welcoming environment. This
+includes but is not limited to use of overtly sexual, vulgar, or otherwise
+offensive nicknames or avatar images.
```
I think having it in the list below is enough. Too much here and we detract from the list. Plus, the more specific we are, the easier it is to circumvent.

tannewt commented on code-of-conduct.md:
```diff
@@ -0,0 +1,165 @@
+include:
+
+* Be kind and courteous to others
+* Behaving professionally
```
What does this cover that isn't covered by the other bullet points?

tannewt replied:
Some people in our spaces may never have worked in a professional setting and may not understand this.

tannewt commented on code-of-conduct.md:
```diff
@@ -0,0 +1,165 @@
+  advances
+* The use of inappropriate images in a user's avatar or inappropriate nicknames
+* Any spamming, flaming, baiting or other attention-stealing behavior
+* Excessive helping
```
Maybe "Excessive and unwelcome helping" to make it a little clearer

tannewt commented on code-of-conduct.md:
```diff
@@ -0,0 +1,165 @@
+an official project e-mail address, posting via an official social media
+account, or acting as an appointed representative at an online or offline event.
+Representation of a project may be further defined and clarified by project
+maintainers.
```
I think these last two sentences are too specific. Actions should be taken on inappropriate behavior relating to CircuitPython even if the person has no official representative designation.

Andon-A reviewed (commented):
Line 19: I'd recommend removing the word "Overtly" as realistically, we don't want the subtle stuff either. And removing the potential for people to claim "It wasn't overt!" and justify their actions to themselves seems good to me.

Line 45: I recommend removing the word "Unwelcome" - Adafruit & CircuitPython spaces are not an appropriate place for sexual attention or advances, wanted or not. 

Line 47: I recommend something along the lines of "The use of inappropriate images, including in a user's avatar, or inappropriate nicknames" to explicitly prohibit *all* inappropriate images.

Line 55: I recommend adding "or public" as some people's professional settings can be significantly more lax. Especially in regards to jokes and conversations that aren't suitable for children.

Line 141: Perhaps mention the Adafruit Be Excellent policy covering other Adafruit things, and link to it?

tannewt commented on code-of-conduct.md:
```diff
@@ -0,0 +1,165 @@
+oppressive, or exclusionary, are not allowed.
+2. Messages, comments, nicknames or avatar images that moderators find
+inappropriate, whether listed in the code of conduct or not, are also not
+allowed.
```
I'd omit these two bullet points and instead start with a "report" step merged from enforcement. Its confusing to list inappropriate behavior in multiple spots.

tannewt commented on code-of-conduct.md:
```diff
@@ -0,0 +1,165 @@
+incident. Be aware, if you send a message in an open channel, your report will
+not be anonymous. In situations where the issue is particularly egregious,
+requires immediate action, and no moderators are available, you can also report
+the message directly to Discord.
```
How splitting it into a smaller "overall" section that covers the support email and then subsections for specific spaces like GitHub and Discord? That should make it more concise and easier to skim.

I think then you can merge this section with the following moderation section to cover the whole enforcement process from report to ban.

tannewt commented on code-of-conduct.md:
```diff
@@ -0,0 +1,165 @@
+official CircuitPython spaces (both public and private), the entire Adafruit
+Discord server, and CircuitPython GitHub repositories under Adafruit. Examples
+of official CircuitPython spaces include meet-ups, audio chats on the Adafruit
+Discord, or interaction at a conference.
```
I don't like limiting to official spaces. Any inappropriate behavior can have repercussions.

tannewt commented on code-of-conduct.md:
```diff
@@ -0,0 +1,165 @@
+other community members can have a much more significant impact than simply
+providing the correct answer.
+
+## Scope
```
I'd move this above enforcement and moderation to set context for where it applies. I think its worth talking about two different scopes as well. The scope of where the CoC applies and the scope of where enforcement applies. (An action in the arduino server that violates our code of conduct can result in a ban from the adafruit server for example.)

tannewt reviewed (changes requested):
Overall, I like the content additions. I worry that many of the new sections are too verbose to be read often. Making it more concise will improve readability and leave us more flexibility in implementation.

Andon-A commented on code-of-conduct.md:
```diff
@@ -0,0 +1,165 @@
+
+Examples of unacceptable behavior by participants include:
+
+* The use of sexualized language or imagery and unwelcome sexual attention or
```
Recommend removing "unwelcome" as Adafruit & CircuitPython spaces are not an appropriate place for sexual attention or advances, wanted or not.

Andon-A commented on code-of-conduct.md:
```diff
@@ -0,0 +1,165 @@
+
+On the Adafruit Discord server, please avoid using nicknames or avatar images
+that might detract from a friendly, safe and welcoming environment. This
+includes but is not limited to use of overtly sexual, vulgar, or otherwise
```
Recommend removing the word "Overtly" as, realistically, we don't want subtle stuff either. And it also removes the potential for people to justify things to themselves by claiming "It wasn't overt!"

Andon-A commented on code-of-conduct.md:
```diff
@@ -0,0 +1,165 @@
+
+* The use of sexualized language or imagery and unwelcome sexual attention or
+  advances
+* The use of inappropriate images in a user's avatar or inappropriate nicknames
```
Recommend changing to "The use of imapproriate images or language, including in a user's avatar or nickname"

This explicitly prohibits *all* usage.

Andon-A commented on code-of-conduct.md:
```diff
@@ -0,0 +1,165 @@
+* Publishing others' private information, such as a physical or electronic
+  address, without explicit permission
+* Other conduct which could reasonably be considered inappropriate in a
+  professional setting
```
Recommend "professional or public setting" - As @tannewt mentions in his review, not everyone has been in a professional setting (IE, students/kids). And not every "professional" setting is the same, especially those where children are far removed.

Andon-A commented on code-of-conduct.md:
```diff
@@ -0,0 +1,165 @@
+This Code of Conduct and the enforcement policies listed above apply to all
+official CircuitPython venues. This includes but is not limited to any
+official CircuitPython spaces (both public and private), the entire Adafruit
+Discord server, and CircuitPython GitHub repositories under Adafruit. Examples
```
Might be worth mentioning and linking the Be Excellent policy which governs Adafruit non-CircuitPython spaces.

Andon-A reviewed (changes requested):
Changed review type as requested. Mostly pedantic changes.

kattni committed:
Updates per PR suggestions

tannewt commented on code-of-conduct.md:
```diff
@@ -14,45 +14,35 @@ race, religion, or sexual identity and orientation.
+We value many things beyond technical expertise, including collaboration and
+supporting others within our community. Providing a positive experience for
+other community members can have a much more significant impact than simply
+providing the correct answer.
```
Yes! Love this section! "answering outside the scope of the question asked" is perfectly worded

kattni replied:
Thank you!

NonsenseInc replied:
I have to disagree with @tannewt. While I assume to understand the intend of the sentence "Excessive or unwelcome helping; answering outside the scope of the question asked", I have to disagree with it's implications; sometimes it seems sensible to me to give the mostly unwanted piece of advice, that "you should not be doing this".

kattni replied:
@NonsenseInc Thank you so much for the feedback!

There are absolutely situations where that fits! Giving the advice of "You should not be doing this" is not considered answering outside the scope of the question asked. If applicable, it's providing good advice within the scope of the question asked. What we're trying to avoid here is the person answering overwhelming the person asking with information outside what they asked, or taking over the conversation to the point where the person asking is no longer really driving it. Initially we had the phrase "taking the keyboard away from" which is a reference to being in person, trying to assist someone with an issue on their own computer, and instead of helping them through the issue, you take the keyboard away from them and do the fix yourself. We didn't want to have an entire paragraph specific to explaining what we mean by excessive helping because we thought it made it seem more important than the other examples of inappropriate behaviors. The concept needs to be included because unfortunately there are people who engage in this behavior, and we want to make sure that it's clear that it is inappropriate.

I agree entirely with your sentiment, but I don't think this wording applies to your example.

NonsenseInc replied:
@kattni Thank you very much for your explaination! In conclusion I think I misunderstood the rule and now agree with it and it's wording.

tannewt commented on code-of-conduct.md:
```diff
@@ -66,42 +56,39 @@ that are not aligned to this Code of Conduct, or to ban temporarily or
+@tannewt#4653, @Dan Halbert#1614, @cater#2442, @sommersoft#0222, or @Andon#8175.
+
+In situations on Discord where the issue is particularly egregious, requires
+immediate action, and no moderators are available, you can also report the
```
I'd say "should" instead of "can". Maybe "particularly egregious, possibly illegal, requires immediate action or violates the Discord terms of service, you should also report the message directly to Discord."

kattni replied:
Agreed. I'll reword this.

tannewt commented on code-of-conduct.md:
```diff
@@ -66,42 +56,39 @@ that are not aligned to this Code of Conduct, or to ban temporarily or
+immediate action, and no moderators are available, you can also report the
+message directly to Discord.
+
+These are the steps for upholding our community’s standards of conduct.
```
This section below still feels out of place to me because its relatively long. I think it'd be better as a pinned post for moderators to follow or separate "moderator how-to" doc that this can link to.

I think the key takeaway for community folks is that for less egregious actions a moderator will reach out privately to involved parties, discuss the issue and make a decision based on the response of folks and after consulting with at least one other moderator as a sanity check. Its important for everyone involved to be open to feedback and be willing to change their behavior.

kattni replied:
I agree that that is the takeaway, however I don't agree that the section should be removed. I can definitely work through it to tighten it up a bit.

The outline of this section is pulled from Rust's CoC. I know we don't want to get too specific on what is and isn't allowed because then we're limiting our scope, but I feel like we should be more specific when it comes to the moderation steps. We're not limiting ourselves with the way this is written, and we're making it entirely clear what will happen if someone violates the CoC. Right now, we don't say to people on their first warning "This is your first warning, depending on how you act you might get a couple more". We simply engage the individual and then keep an eye on them moving forward. I feel like it's an important thing to include so people know the process. Codes of Conduct aren't necessarily there for the people who are following them, they're there for the people who aren't. And those are the people who would need to have the moderation steps laid out for them.

tannewt reviewed (changes requested):
I'm really excited about this! Just a bit more feedback on the second half.

kattni committed:
Updates to moderation

sommersoft commented on code-of-conduct.md:
```diff
@@ -0,0 +1,126 @@
+3. We have a soft "three strikes" policy - we are willing to give a user the
+chance to change their behavior. However, how the user respond to the
+warning is important - it will determine whether or not the user receives
+further chances.
```
The wording on this kind of concerns me. I fear it can be interpreted in a "I can make a violation, then attempt to talk my way out of it". I think its a good thing making it known that reaction(s) to a "strike" carry weight. It might be the "further chances" part that is flagging for me. 

I'll formulate my thoughts a little more after sleep (just got back home).

kattni replied:
The next bit in the list states "If the user is receptive to the warning and changes their behavior..." We're already explaining what is necessary to be allowed to continue. And the fact is, this is exactly what we're doing right now. So yes, theoretically someone could be making a violation and trying to talk their way out of it, but I guess if that's someone's goal in life, to violate the Adafruit CP CoC once and then change their behavior, they're going to do it and "get away" with it once. 

Further, there have been a few people who seem to operate under that assumption, violate and talk their way out of it, and after three times, we've banned them. So I think the explanation is pretty clear and the bits that follow solidify it.

tannewt replied:
I think this can be folded into 4 instead of being on its own by adding a caveat for repeated warnings. 2 is "warn the user", 4 and 5 (would be 3 and 4) are how the person responds.

sommersoft commented on code-of-conduct.md:
```diff
@@ -0,0 +1,126 @@
+4. If the user is receptive to the warning and changes their behavior, they 
+will be given another chance.
+5. If the user is unreceptive or unreasonable when warned by a moderator, or
+the warning goes unheeded, they may be banned for a first or second offense.
```
Should 4 & 5 be subordinate to 3 (i.e. 3a & 3b)? They don't stand on their own, without the previous "three strikes" information.

And yes...pedantic. :smile:

sommersoft replied:
Actually, in re-reading, they are more subordinate to paragraph 2's warning info.

kattni replied:
I think this is fine as it is. I've already taken out half of what was there. We want out policy outlined but I think getting into 3a. and 3b. starts detracting from the point of the list. It's meant to be something to read through so you know the process, not to be a perfectly designed legal document.

tannewt commented on code-of-conduct.md:
```diff
@@ -0,0 +1,126 @@
+behavior and are expected to take appropriate and fair corrective action in
+response to any instances of unacceptable behavior.
+
+Project maintainers have the right and responsibility to remove, edit, or
```
Maybe "leaders" instead of "maintainers" to distance it from code only areas?

tannewt commented on code-of-conduct.md:
```diff
@@ -0,0 +1,126 @@
+1. Any member of the community may report any situation that violates the
+Adafruit CircuitPython Code of Conduct. All reports will be reviewed and 
+investigated, and will result in a response that is deemed necessary and 
+appropriate to the circumstances.
```
I don't think the second sentence is needed. Its covered in Our Responsibilities.

tannewt commented on code-of-conduct.md:
```diff
@@ -0,0 +1,126 @@
+5. If the user is unreceptive or unreasonable when warned by a moderator, or
+the warning goes unheeded, they may be banned for a first or second offense.
+6. If the behavior is an egregious violation, the user may be banned
+immediately, without warning.
```
I think this should be 2 since its a quick resolution to a raised issue.

tannewt reviewed (changes requested):
Getting closer! Looking really good. Do we want broader review?

kattni committed:
Streamlined Moderation steps, changed "user" to "community member"

User was a carry over from adapting Rust's IRC guidelines. It was
not used consistently, and did not lend itself to the scope of our
CoC. Updated all instances of "user" to "community member" to
better reflect our scope.

Also changed wording in second step in Moderation to reflect that
the banning applies to the community member who committed the
violation. This seems obvious, but moving the immediate ban step
to below the reporting step meant that it could have read as the
community member who reported the violation may be banned.

Removed use of "We" from Moderation steps as it was inconsistent
with the rest of the CoC.

kattni committed:
Streamlined Moderation steps, changed "user" to "community member"

User was a carry over from adapting Rust's IRC guidelines. It was
not used consistently, and did not lend itself to the scope of our
CoC. Updated all instances of "user" to "community member" to
better reflect our scope.

Also changed wording in second step in Moderation to reflect that
the banning applies to the community member who committed the
violation. This seems obvious, but moving the immediate ban step
to below the reporting step meant that it could have read as the
community member who reported the violation may be banned.

Removed use of "We" from Moderation steps as it was inconsistent
with the rest of the CoC.

tannewt reviewed (approved):
Looks awesome! Thank you for doing these updates!

myersjustinc commented:
I don't have any problem at all with this code of conduct, but it's unclear to me what the relationship is between CircuitPython and the Adafruit Discord. (I'm not even really clear on what CircuitPython _is_, to be honest.)

I mainly just lurk on the Discord server, and I never once understood myself to be doing something CircuitPython-related—so if I saw a document billing itself as a "CircuitPython Code of Conduct", I wouldn't think to read it because I wouldn't think it applies to me. (Only heard about it today because `@everyone` got pinged on Discord.) And if I _had_ started, it doesn't help that the only mentions of Discord are in the last couple of sections, after I otherwise might have stopped reading.

Is there a way to make it more obvious that this also applies to Discord and the people who only interact with the community over there?

daniel-stewart reviewed (commented):
Overall, I really like it. I would suggest ending the "Our Pledge" section (line 7 of code of conduct.md) with "harrassment-free for everyone." I believe the simplicity of this should be enough. Following it with a list of 13 categories seems to me to diminish the statement that it is for everyone. It's for everyone, full stop.

NonsenseInc commented on code-of-conduct.md:
```diff
@@ -0,0 +1,124 @@
+reject messages, comments, commits, code, issues, and other contributions
+that are not aligned to this Code of Conduct, or to ban temporarily or
+permanently any community member for other behaviors that they deem
+inappropriate, threatening, offensive, or harmful.
```
I would like to suggest, that any editing or removing done by moderation should have to be indicated to prevent misrepresentation and accusations of such.

dhalbert reviewed (approved):
Great! Looking forward to citing this.

kattni committed:
Merge branch 'master' into code-of-conduct

kattni committed:
Updates per community feedback

kattni commented:
@daniel-stewart Thank you for the suggestion! I agree the simplicity should be enough, however my concern is that in some situations, it has proven not to be. I understand where you're coming from with this, but I think we're going to leave it as-is for now.

kattni commented:
@myersjustinc Thank you for the feedback! I agree it was written to be more specific than it should have been, and needed to include a broader scope. We have generalised the Code of Conduct to be for the Adafruit Community, which includes the Discord Community.

tannewt commented on code-of-conduct.md:
```diff
@@ -0,0 +1,127 @@
+<https://www.contributor-covenant.org/version/1/4/code-of-conduct.html>,
+and the [Rust Code of Conduct](https://www.rust-lang.org/en-US/conduct.html).
+
+For other projects adopting the Adafruit CircuitPython Community Code of
```
Drop the CircuitPython here.

kattni replied:
Thanks for catching this!

tannewt reviewed (approved):
Looks awesome! One minor thing and then it looks ready to go. Thank you so much for refining this! Lets plan on rolling it out next week to Discord and GitHub repos.

kattni committed:
Removed errant CircuitPython

kattni commented:
Sounds like an excellent plan!

dhalbert reviewed (approved):
Thanks for your extensive work on this!
